x-pack/plugins/apm/server/lib/service_map/get_service_map_service_node_info.ts

```diff
@@ -40,66 +45,64 @@ export async function getServiceMapServiceNodeInfo({
   const filter: ESFilter[] = [
     { range: rangeFilter(start, end) },
     { term: { [SERVICE_NAME]: serviceName } },
-    ...(environment ? [{ term: { [SERVICE_ENVIRONMENT]: environment } }] : []),
   ];
 
+  const environmentFilter = getEnvironmentUiFilterES(environment);
+
+  if (environmentFilter) {
+    filter.push(environmentFilter);
+  }
```

Comment:
I don't think we need to do this - and afaict `environment` should be removed from the signature.
`setup` already contains `uiFilters` which contains `environment`. 

So all that's needed is to forward `uiFilters` `getErrorStats`

Reply:
But `setup` does not contain `uiFilters`. I can change it so it does.

Reply:
In this case I don't think we want `environment` from `uiFilters` anyway, since we should be using the environment for the selected node showing the popover and not for the environment selector, which `uiFilters` would hold.

Reply:
> since we should be using the environment for the selected node showing the popover and not for the environment selector, which uiFilters would hold.

Okay, that makes sense. Thanks for clarifying - was not aware of this difference.